blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/EnumMetricDataIO.java

```diff
@@ -0,0 +1,199 @@
+     *
+     * @param locators
+     * @param range
+     * @param gran
+     * @return
+     */
+    public Map<Locator, MetricData> getEnumMetricDataForRangeForLocatorList(final List<Locator> locators, final Range range, final Granularity gran) {
+        String columnFamily = CassandraModel.getColumnFamily(BluefloodEnumRollup.class, gran).getName();
+        if ( columnFamily == null ) {
+            throw new IllegalArgumentException("invalid granularity: " + gran);
+        }
+        return getEnumMetricDataForRangeForLocatorList(locators, range, columnFamily);
+    }
+
+    /**
+     * Reads from both the metrics_enum and metrics_preaggregated_{granularity} column families,
+     * in parallel, and join the data to construct {@link com.rackspacecloud.blueflood.outputs.formats.MetricData}
+     * objects
+     *
+     * @param locators
+     * @param range
+     * @param columnFamily
+     * @return
+     */
+    private  Map<Locator, MetricData> getEnumMetricDataForRangeForLocatorList(final List<Locator> locators, final Range range, final String columnFamily) {
+
+        if (range.getStart() > range.getStop()) {
+            throw new IllegalArgumentException(String.format("invalid range: ", range.toString()));
+        }
+
+        Future<Table<Locator, Long, String>> enumHashValuesFuture = taskExecutor.submit(new Callable() {
+            @Override
+            public Table<Locator, Long, String> call() throws Exception {
+                return enumReader.getEnumHashValuesForLocators(locators);
+            }
+
+        });
+
+        Future<Table<Locator, Long, BluefloodEnumRollup>> enumHashRollupFuture = taskExecutor.submit(new Callable() {
+            @Override
+            public Table<Locator, Long, BluefloodEnumRollup> call() throws Exception {
+                return enumReader.getEnumRollupsForLocators(locators, columnFamily, range);
+            }
+        });
+
+        Map<Locator, MetricData> resultMap = new HashMap<Locator, MetricData>();
+        try {
+            Table<Locator, Long, String> enumHashValues =  enumHashValuesFuture.get();
+            Table<Locator, Long, BluefloodEnumRollup> enumHashRollup = enumHashRollupFuture.get();
+
+            for (Locator locator : locators) {
+                populateEnumValueToCountMap(enumHashRollup.row(locator), enumHashValues.row(locator));
+                Points points = convertToPoints(enumHashRollup.row(locator));
+                MetricData metricData = new MetricData(points, getUnitString(locator), MetricData.Type.ENUM);
+                resultMap.put(locator, metricData);
+            }
+        } catch (InterruptedException e) {
+            LOG.error("Interrupted while reading Enum metrics for locators " + locators, e);
+        } catch (ExecutionException e) {
+            LOG.error("Execution error while reading Enum metrics for locators " + locators, e);
+        }
```

Comment:
I assume ignoring the error and continuing was what we did before.  Do we want to make a story to go back and think how best to handle errors and what to return the consumer?